integration-tests/api-gateway/endpoints.ts

```diff
@@ -1,124 +1,65 @@
-import { getSSMParameter } from "../resources/ssm-param-helper";
-import {
-  Payload,
-  getJarAuthorizationPayload,
-} from "./crypto/create-jar-request-payload";
-import {
-  getClaimSet,
-  CLIENT_ID,
-  CLIENT_URL,
-  environment,
-} from "./env-variables";
+import { environment, testResourcesStack } from "./env-variables";
 import { stackOutputs } from "../resources/cloudformation-helper";
-
-let publicEncryptionKeyBase64: string;
-let privateSigningKey: any;
-let privateAPI: string;
-let preOutput: Partial<{
-  PrivateApiGatewayId: string;
-}>;
-
-type Name = {
-  name: {
-    nameParts: {
-      type: string;
-      value: string;
-    }[];
-  }[];
-};
-
-export const createPayload = async (sharedClaimsUpdate?: Name) => {
-  publicEncryptionKeyBase64 =
-    (await getSSMParameter(
-      "/check-hmrc-cri-api/test/publicEncryptionKeyBase64"
-    )) || "";
-  privateSigningKey = JSON.parse(
-    (await getSSMParameter("/check-hmrc-cri-api/test/privateSigningKey")) || ""
-  );
-  preOutput = await stackOutputs(process.env.STACK_NAME);
-  privateAPI = `${preOutput.PrivateApiGatewayId}`;
-  const correctClaimSet = await getClaimSet();
-  const updateClaimset = {
-    ...correctClaimSet,
-    ...sharedClaimsUpdate,
-    name: sharedClaimsUpdate?.name || correctClaimSet.shared_claims.name,
-  };
-  const audience = correctClaimSet.aud;
-  const payload = {
-    clientId: CLIENT_ID,
-    audience,
-    authorizationEndpoint: `${audience}/oauth2/authorize`,
-    redirectUrl: `${CLIENT_URL}/callback`,
-    publicEncryptionKeyBase64: publicEncryptionKeyBase64,
-    privateSigningKey: privateSigningKey,
-    issuer: CLIENT_URL,
-    claimSet: updateClaimset,
-  } as unknown as Payload;
-  const ipvCoreAuthorizationUrl = await getJarAuthorizationPayload(payload);
-  return ipvCoreAuthorizationUrl;
-};
-
-export const createSession = async (): Promise<Response> => {
-  const ipvCoreAuthorizationUrl = await createPayload();
-  const sessionApiUrl = `https://${privateAPI}.execute-api.eu-west-2.amazonaws.com/${environment}/session`;
-  const sessionResponse = await fetch(sessionApiUrl, {
-    method: "POST",
-    headers: {
-      "Content-Type": "application/json",
-      "X-Forwarded-For": "localhost",
-    },
-    body: JSON.stringify(ipvCoreAuthorizationUrl),
-  });
-
-  return sessionResponse;
+import { signedFetch } from "../resources/fetch";
+import { JWTClaimsSet } from "./types";
+
+type JarAuthorizationOptions = {
+  clientId?: string;
+  aud?: string;
+  iss?: string;
+  claimsOverride?: unknown;
+  evidence_requested?: unknown;
 };
-
-export const createInvalidSession = async (): Promise<Response> => {
-  const ipvCoreAuthorizationUrl = await createPayload();
-  const sessionApiUrl = `https://${privateAPI}.execute-api.eu-west-2.amazonaws.com/${environment}/session`;
-  const sessionResponse = await fetch(sessionApiUrl, {
+export const getJarAuthorization = async ({
+  clientId,
+  aud,
+  iss,
+  claimsOverride,
+  evidence_requested,
```

Comment:
[nit] can we please be consistent with casing - think this needs to be camelCase